src/components/orders/FilledProgress/index.tsx

```diff
@@ -114,70 +116,39 @@ export function FilledProgress(props: Props): JSX.Element {
   } = props
 
   const touched = filledPercentage.gt(0)
-
-  let mainToken
-  let mainAddress
-  let mainAmount
-  let swappedToken
-  let swappedAddress
-  let swappedAmount
-  let action: string | null | undefined
-
-  let filledAmountWithFee, swappedAmountWithFee
-  if (kind === 'sell') {
-    action = 'sold'
-
-    mainToken = sellToken
-    mainAddress = sellTokenAddress
-    mainAmount = sellAmount.plus(feeAmount)
-
-    swappedToken = buyToken
-    swappedAddress = buyTokenAddress
-    swappedAmount = executedBuyAmount
-
-    // Sell orders, add the fee in to the sellAmount (mainAmount, in this case)
-    filledAmountWithFee = filledAmount.plus(executedFeeAmount)
-    swappedAmountWithFee = swappedAmount
-  } else {
-    action = 'bought'
-
-    mainToken = buyToken
-    mainAddress = buyTokenAddress
-    mainAmount = buyAmount
-
-    swappedToken = sellToken
-    swappedAddress = sellTokenAddress
-    swappedAmount = executedSellAmount
-
-    // Buy orders need to add the fee, to the sellToken too (swappedAmount in this case)
-    filledAmountWithFee = filledAmount
-    swappedAmountWithFee = swappedAmount.plus(executedFeeAmount)
-  }
+  const isSell = kind === 'sell'
+
+  const mainToken = (isSell ? sellToken : buyToken) || null
```

Comment:
you really hate some much `let` than you added a bit of repetition. Agree on killing all those bad  lets, but why not doing a helper function with

```
if (isSell)
    return {  ... }
else 
    return {  ... }

```

Reply:
I just wanna have this code compact, before it took 38 lines of code, now only 12

Reply:
I treat it as a table:
| - | - | - | - |
|--------|--------|--------|--------|
| const mainToken = | isSell ? | sellToken | buyToken |
| const mainAddress = | isSell ? | sellTokenAddress | buyTokenAddress |